src/solver/hydro/management/management.cpp

```diff
@@ -135,6 +137,89 @@ void HydroManagement::prepareInflowsScaling(uint numSpace)
             {
                 data.inflows[realmonth] = totalMonthInflows;
             }
+
+            //CR22: Monthly minimum generation <= Monthly inflows for each month
+            double totalMonthMingen = 0.0;
+            for (uint d = firstDayOfMonth; d != firstDayOfNextMonth; ++d)
+            {
+                for (uint h = 0; h < 24; ++h)
+                {
+                    totalMonthMingen += srcmingen[d*24 + h];
+                }          
+            }
+
+            //CR22: set montly mingen, used later for h2o_m
+            if (not(area.hydro.reservoirCapacity < 1e-4))
+            {
+                if (area.hydro.reservoirManagement)
+                {
+                    data.mingens[realmonth] = totalMonthMingen / (area.hydro.reservoirCapacity);
+                    assert(!Math::NaN(data.mingens[month]) && "nan value detect in mingen");
+                }
+                else
+                    data.mingens[realmonth] = totalMonthMingen;
+            }
+            else
+            {
+                data.mingens[realmonth] = totalMonthMingen;
+            }                  
+
+            if (area.hydro.followLoadModulations and not area.hydro.reservoirManagement)
+            {
+                if(totalMonthMingen > totalMonthInflows)
+                {
+                    logs.error() << "In Area "<< area.name << " the minimum generation of "
+                    << totalMonthMingen << " MW in month " << month + 1 << " of TS-" << tsIndex + 1<< " is incompatible with the inflows of "
+                    << totalMonthInflows << " MW.";
+                }
+            }
+        }
+
+        if (area.hydro.followLoadModulations and area.hydro.reservoirManagement)
+        {
+            //CR22: Yearly minimum generation <= Yearly inflows for each year 
+            double totalYearMingen = 0.0;
+            double totalYearInflows = 0.0;
+            for (uint hour = 0; hour < HOURS_PER_YEAR; ++hour)
+            {
+                totalYearMingen += srcmingen[hour];
+            }
+            for (uint day = 0; day < DAYS_PER_YEAR; ++day)
+            {
+                totalYearInflows += srcinflows[day];
+            }
+            if(totalYearMingen > totalYearInflows)
+            {
+                logs.error() << "In Area "<< area.name << " the minimum generation of "
+                << totalYearMingen << " MW of TS-" << tsIndex + 1 << " is incompatible with the inflows of "
+                << totalYearInflows << " MW.";
+            }
+        }
+
+        if (not area.hydro.followLoadModulations)
+        {
+            //CR22: Weekly minimum generation <= Weekly inflows for each week 
+            for (uint week = 0; week < 53; ++week)
```

Comment:
52 ?

Reply:
52 weeks x 168 h/week = 8736 h
we have HOURS_PER_YEAR = 8760 entries.
If we go just up to 52 weeks we "forget" the last 8760-8736=24h
so it's 53 weeks but with the additional condition " if (hour < HOURS_PER_YEAR)" not to go above 8760 and sum some non-existing values.

Reply:
I made some digging on this.

_A bit of context on the calendar_ :
The reason why we may have 366 days (and therefore 8760 hours) in a year of the calendar is that we may run a simulation on a leap year (see **GUI > Simulation > Calendar > Leap year**).
So we can have a simulation of many MC leap years.
But the simulation will always run on 52 weeks maximum, this means there is a day for which we don't need the input data (data related to this day in the time series), even if we load the whole TS.

That said, in the hydro heuristic like here, you have to deal with the whole calendar 
(whatever the duration of a simulation and whether it runs on a leap year or not), right ?

- You're right to loop over 53 weeks. But I would recommend to use `calendar.maxWeeksInYear`. For some reason I don't know, `calendar.maxWeeksInYear` = 54. So, instead of 53, you may use `calendar.maxWeeksInYear - 1`
- In the calendar, the last week contains only one day, that is 24 hours, so **hour** (index of hours) will not exceed **HOURS_PER_YEAR** in the last week.

To sum up, I recommend : 
```C++
for(uint hour = calendar.weeks[week].hours.first; hour < calendar.weeks[week].hours.end; ++hour)
 {
     totalWeekMingen += srcmingen[hour];
 }
```
An you could also make use of **std::accumulate** instead
```C++
double totalWeekMingen = 0.0;
totalWeekMingen = std::accumulate(srcmingen + calendar.weeks[week].hours.first,
                                  srcmingen + calendar.weeks[week].hours.end,
                                  totalWeekMingen);
```

Reply:
I made some digging on this.

_A bit of context on the calendar_ :
The reason why we may have 366 days (and therefore 8760 hours) in a year of the calendar is that we may run a simulation on a leap year (see **GUI > Simulation > Calendar > Leap year**).
So we can have a simulation of many MC leap years.
But the simulation will always run on 52 weeks maximum, this means there is a day for which we don't need the input data (data related to this day in the time series), even if we load the whole TS.

That said, in the hydro heuristic like here, you have to deal with the whole calendar 
(whatever the duration of a simulation and whether it runs on a leap year or not), right ?

- You're right to loop over 53 weeks. But I would recommend to use `calendar.maxWeeksInYear`. For some reason I don't know, `calendar.maxWeeksInYear` = 54. So, instead of 53, you may use `calendar.maxWeeksInYear - 1`
- In the calendar, the last week contains only one day, that is 24 hours, so **hour** (index of hours) will not exceed **HOURS_PER_YEAR** in the last week.

To sum up, I recommend : 
```C++
for(uint hour = calendar.weeks[week].hours.first; hour < calendar.weeks[week].hours.end; ++hour)
 {
     totalWeekMingen += srcmingen[hour];
 }
```
An you could also make use of **std::accumulate** instead
```C++
double totalWeekMingen = 0.0;
totalWeekMingen = std::accumulate(srcmingen + calendar.weeks[week].hours.first,
                                  srcmingen + calendar.weeks[week].hours.end,
                                  totalWeekMingen);
```

Reply:
I agree with "calendar.maxWeeksInYear - 1". Will be implemented.
For the second part. In theory, this should work. We implemented it like this the first time. 
But during testing this happens:
![image](https://user-images.githubusercontent.com/97689304/229437369-999bb41c-6462-4039-ae99-5f6c0492950e.png)

And you end up with dummy errors:
![image](https://user-images.githubusercontent.com/97689304/229437505-bdb7d6bf-7011-4ed7-a68b-a25d3b63e633.png)

So what happens? The hourly iterations continue after 8760. And to make things worse starts to sum up some crazy values for:
srcinflows[hour]
Actually, I realized these crazy values are numbers from week-0, hours-0-1-2..... 
It basically makes a full circle and starts summing from the start again and putting that in week 52. 
So:
if (hour < HOURS_PER_YEAR)
is kinda unavoidable here.